BitwardenShared/UI/Platform/Application/AppProcessor.swift

```diff
@@ -383,14 +383,27 @@ extension AppProcessor {
     /// - Returns: an `AppRoute` if one was successfully built from the URL passed in, `nil` if not.
     ///
     private func getBitwardenUrlRoute(url: URL) async -> AppRoute? {
-        guard let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
-              let scheme = components.scheme,
-              scheme.isBitwardenAppScheme,
-              components.host == "settings",
-              components.path == "/account_security"
-        else { return nil }
-
-        return AppRoute.tab(.settings(.accountSecurity))
+        guard let scheme = url.scheme, scheme.isBitwardenAppScheme else { return nil }
+
+        var route: AppRoute?
+        switch url.absoluteString {
+        case IncomingLinkConstants.accountSecurity:
+            route = AppRoute.tab(.settings(.accountSecurity))
+        case IncomingLinkConstants.authenticatorNewItem:
+            guard let item = await services.authenticatorSyncService?.getTemporaryTotpItem(),
+                  let totpKey = item.totpKey,
+                  let otpAuthModel = OTPAuthModel(otpAuthKey: totpKey) else {
+                coordinator?.showAlert(.defaultAlert(title: Localizations.somethingWentWrong,
+                                                     message: Localizations.pleaseTryAgain))
```

Comment:
🤔  Is it possible to have a more specific message to the user as to what went wrong? Would trying again actually fix this somehow at any moment for the same `totpKey`?

Reply:
I talked this over with Emily and she had the same question 😆 This is the error we landed on - while trying again won't likely solve the problem, this really shouldn't have happened. If you came here via the correct deeplink and the Authenticator app is storing it correctly in the store, it should be there when we fetch it.

The one exception to this is the known issue I listed above which would be addressed by [BITAU-179](https://livefront.atlassian.net/browse/BITAU-179)

Reply:
Alright, if this is only possible from the Authenticator app then it's highly unlikely to happen but if any other app can launch this URL I believe we should improve the messaging to state something about the OTP format being wrong or something like that.

Reply:
This particular url will only ever work with the Authenticator app because it depends on an item being added first to the temporary store (which only the Authenticator app can do). The deeplink serves as a way to launch the PM app and tell it to check for the new item (i.e. it's not providing any info via the deeplink like you might from a different app or the camera).

Reply:
Indeed but I was thinking about the **non-happy path** where some app or something fires this URL and is opened by Bitwarden and given the temp item won't exist the user will see _"Something went wrong, please try again"_ with no idea about why that is shown. Again not 100% sure if any other app can fire this Deep link, can't remember right now if it's somehow configured to work only inside our AppGroup or something like that.

Reply:
@fedemkr I talked through this with Emily and she suggested two possible alternatives (keeping the headline of "Something went wrong" the same):
- "Unable to open the selected item. Please try again."
- "Unable to move the selected item to Bitwarden. Please try again."

We both lean towards the first one. Thoughts?

Reply:
@brant-livefront talked it with Micah and he said if we could know the source application that fired the deep link. I've checked and there's a [sourceApplication](https://developer.apple.com/documentation/uikit/uiscene/openurloptions) in the url options of the scene `urlContexts` options which I think it's exactly what we want.
Also, mentioned the next thing:
> If it turns out that we can verify the source of the request as the authenticator app, use this copy:
“Unable to move the selected item from the Bitwarden Authenticator. Please try again.”
If we can’t, then I would use this copy:
“Unable to move the selected item. Please try again.”

Reply:
@fedemkr I've implemented this locally, but I'm not sure it's worth all the added complexity - we need to fetch this in SceneDelegate, pass down to AppProcessor.openUrl, pass through to getBitwardenRoute and then we can do the check:
``` Swift
                let message = source == BitwardenDeepLinkConstants.authenticatorBundleId ?
                    Localizations.unableToMoveTheSelectedItemFromTheBitwardenAuthenticatorPleaseTryAgain :
                    Localizations.unableToMoveTheSelectedItemPleaseTryAgain

                coordinator?.showAlert(.defaultAlert(title: Localizations.somethingWentWrong,
                                                     message: message))
```

But that just gives us a slightly more verbose error message when we verify the source of the link? Am I understanding that correctly? 

Emily and I both lean towards using the "Unable to move the selected item. Please try again.” copy in both cases. If you and Micah feel strongly though, I can push up a commit that adds the additional source check.

Reply:
@brant-livefront Micah agrees to go with your suggestion on the generic message for both cases, so no need to validate the source. Thanks for all the feedback and sorry for all the back and forth with this.